roles-tag-and-release.sh

```diff
@@ -106,7 +106,17 @@ if [ "$skip" = false ]; then
         ${EDITOR:-vi} "$rel_notes_file"
         if [ -f CHANGELOG.md ]; then
             read -r -p "Edit CHANGELOG.md - press Enter to continue"
-            cat "$rel_notes_file" CHANGELOG.md > .tmp-changelog
+			pat="=*"
```

Comment:
```bash
  myheader="Changelog
========="
  clheader=$(head -2 CHANGELOG.md)
  if [ "$myheader" = "$clheader" ]; then
    echo "$clheader" > .tmp-changelog
    cat "$rel_notes_file" >> .tmp-changelog
    tail -n +3 CHANGELOG.md >> .tmp-changelog
  else
    echo WARNING: Changelog header "$clheader"
    echo not in expected format "$myheader"
    cat "$rel_notes_file" CHANGELOG.md > .tmp-changelog
  fi
```

Reply:
Thanks, @richm. If we are more conscious about the markdown format, `roles-tag-and-release.sh` generates `## [VERSION] - DATE` and stores in `$rel_notes_file`. We still don't know to run `markdownlint.yml` action in each role, but considering the rules set for the network role, we'd like to have `[1.9.4] - 2022-06-28` in the Setext style. Do you think it's possible? (please note that it's ok to have `### New features` and `### Bug fixes`.)

Example of `$rel_notes_file`.
```
## [1.9.4] - 2022-06-28

### New features

- use gather_facts: true in tests_include_vars_from_parent
- support ansible-core-2.13

### Bug fixes

- make min_ansible_version a string in meta/main.yml
```

Reply:
BTW, it's not a good idea to replace the ATX format with Setext before inserting "$rel_notes_file" like this?
```
sed -i -e "s/^# Changelog/Changelog\n=========/" \
       -e "s/^## \(.*\)/\1\n--------------------/" CHANGELOG.md
```
Is it too brutal?